src/io/functions/iologindata_load_player.cpp

```diff
@@ -750,153 +737,134 @@ void IOLoginDataLoad::loadPlayerPreyClass(const std::shared_ptr<Player> &player,
-
-				unsigned long preySize;
-				const char* preyStream = result->getStream("monster_list", preySize);
-				PropStream propPreyStream;
-				propPreyStream.init(preyStream, preySize);
-
-				uint16_t raceId;
-				while (propPreyStream.read<uint16_t>(raceId)) {
-					slot->raceIdList.push_back(raceId);
-				}
-
-				player->setPreySlotClass(slot);
-			} while (result->next());
-		}
+	if (!g_configManager().getBoolean(PREY_ENABLED)) {
+		return;
 	}
-}
 
-void IOLoginDataLoad::loadPlayerTaskHuntingClass(const std::shared_ptr<Player> &player, DBResult_ptr result) {
-	if (!result || !player) {
-		g_logger().warn("[{}] - Player or Result nullptr", __FUNCTION__);
+	std::string query = fmt::format("SELECT * FROM `player_prey` WHERE `player_id` = {}", player->getGUID());
+	result = g_database().storeQuery(query);
+	if (!result) {
 		return;
 	}
 
-	if (g_configManager().getBoolean(TASK_HUNTING_ENABLED)) {
-		Database &db = Database::getInstance();
-		std::ostringstream query;
-		query << "SELECT * FROM `player_taskhunt` WHERE `player_id` = " << player->getGUID();
-		if ((result = db.storeQuery(query.str()))) {
-			do {
-				auto slot = std::make_unique<TaskHuntingSlot>(static_cast<PreySlot_t>(result->getNumber<uint16_t>("slot")));
-				auto state = static_cast<PreyTaskDataState_t>(result->getNumber<uint16_t>("state"));
-				if (slot->id == PreySlot_Two && state == PreyTaskDataState_Locked) {
-					if (!player->isPremium()) {
-						slot->state = PreyTaskDataState_Locked;
-					} else {
-						slot->state = PreyTaskDataState_Selection;
-					}
-				} else {
-					slot->state = state;
-				}
-				slot->selectedRaceId = result->getNumber<uint16_t>("raceid");
-				slot->upgrade = result->getNumber<bool>("upgrade");
-				slot->rarity = static_cast<uint8_t>(result->getNumber<uint16_t>("rarity"));
-				slot->currentKills = result->getNumber<uint16_t>("kills");
-				slot->disabledUntilTimeStamp = result->getNumber<int64_t>("disabled_time");
-				slot->freeRerollTimeStamp = result->getNumber<int64_t>("free_reroll");
-
-				unsigned long taskHuntSize;
-				const char* taskHuntStream = result->getStream("monster_list", taskHuntSize);
-				PropStream propTaskHuntStream;
-				propTaskHuntStream.init(taskHuntStream, taskHuntSize);
-
-				uint16_t raceId;
-				while (propTaskHuntStream.read<uint16_t>(raceId)) {
-					slot->raceIdList.push_back(raceId);
-				}
+	do {
+		auto slot = std::make_unique<PreySlot>(static_cast<PreySlot_t>(result->getNumber<uint16_t>("slot")));
+		auto state = static_cast<PreyDataState_t>(result->getNumber<uint16_t>("state"));
 
-				if (slot->state == PreyTaskDataState_Inactive && slot->disabledUntilTimeStamp < OTSYS_TIME()) {
-					slot->state = PreyTaskDataState_Selection;
-				}
+		if (slot->id == PreySlot_Two && state == PreyDataState_Locked) {
+			if (!player->isPremium()) {
+				slot->state = PreyDataState_Locked;
+			} else {
+				slot->state = PreyDataState_Selection;
+			}
```

Comment:
This will help reduce the conditionals nest complexity.
```suggestion
			slot->state = player->isPremium() ? PreyDataState_Selection : PreyDataState_Locked;
```